modules/ingest-service-impl/src/main/java/org/opencastproject/ingest/impl/IngestServiceImpl.java

```diff
@@ -1349,44 +1351,99 @@ private MediaPackage mergeScheduledMediaPackage(MediaPackage mp) throws IngestEx
     }
   }
 
+  /**
+   * Merge different elements from capture agent ingesting mp and Asset manager. Overwrite or replace same flavored
+   * elements depending on the Ingest Service overwrite configuration. Ignore publications (i.e. live publication
+   * channel from Asset Manager) Always keep tracks from the capture agent.
+   *
+   * @param mp
+   *          the medipackage being ingested from the Capture Agent
+   * @param scheduledMp
+   *          the mediapckage that was schedule and managed by the Asset Manager
+   */
   private void mergeMediaPackageElements(MediaPackage mp, MediaPackage scheduledMp) {
+
     for (MediaPackageElement element : scheduledMp.getElements()) {
-      // Asset manager media package may have a publication element (for live) if retract live has not run yet
-      if (element.getFlavor() != null
-              && !MediaPackageElement.Type.Publication.equals(element.getElementType())
-              && mp.getElementsByFlavor(element.getFlavor()).length > 0) {
-        logger.info("Ignore scheduled element '{}', there is already an ingested element with flavor '{}'", element,
-                element.getFlavor());
+      if (MediaPackageElement.Type.Publication.equals(element.getElementType())) {
+        // The sset managed media package may have a publication element for a live event, if retract live has not run yet.
+        // Publications do not have flavors and are never part of the mediapackage from the capture agent.
+        // Therefore, ignore publication element because it is removed when the recorded media is published and causes complications (on short media) if added.
+        logger.debug("Ignoring {}, not adding to ingested mediapackage {}", MediaPackageElement.Type.Publication, mp);
         continue;
+      } else if (mp.getElementsByFlavor(element.getFlavor()).length > 0) {
+        // The default is to overwrite matching flavored elements in the Asset managed mediapackage (e.g. catalogs)
+        // If isOverwrite is true, changes made from the CA overwrite (update/revert) changes made from the Admin UI.
+        // If isOverwrite is false, changes made from the CA do not overwrite (update/revert) changes made from the Admin UI.
+        // regardless of overwrite, always keep new ingested tracks.
+        if (isOverwrite || MediaPackageElement.Type.Track.equals(element.getElementType())) {
+          // Allow updates made from the Capture Agent to overwrite existing metadata in Opencast
+          logger.debug(
```

Comment:
While I understand the logic here, I can see people being confused by messages about omitting existing elements when it's talking about tracks.

Reply:
Hi @gregorydlogan, actually tracks are always taken from the ingest, it's all the other potential attachments and catalogs that come in on the ingest that are being addressed here, not the tracks.

Reply:
I fixed comments to clarify